parser/encoding/request.go

```diff
@@ -0,0 +1,138 @@
+package encoding
```

Comment:
A few tests would be nice!

Reply:
Indeed! Can I do a follow up PR to not block mr Black?